autotest/ogr/ogr_shape.py

```diff
@@ -4894,6 +4894,69 @@ def test_ogr_shape_write_multipolygon_z_non_finite():
     ogr.GetDriverByName('ESRI Shapefile').DeleteDataSource('/vsimem/test.shp')
 
 ###############################################################################
+# Test that date fields stored as ISO-8601 can be used with IN operator
+
+def test_ogr_in_date_filter():
```

Comment:
As this is not shapefile specific, should rather be moved into autotest/ogr/ogr_sql_rfc28.py